e2e-tests/test/share.test.js

```diff
@@ -0,0 +1,103 @@
+import { ClientFunction, Selector } from 'testcafe'
+import { retrieveEmail } from 'test-common/real-email-config'
+import { waitForReact } from 'testcafe-react-selectors'
+import Page from './PageModels/page-model'
+
+const config = require('../lib/config')
+
+const page = new Page()
+const email = config.getEmail()
+const baseUrl = config.getBaseURL()
+
+fixture(`Login`)
+  .page(baseUrl + '/login')
+  .beforeEach(() => waitForReact())
+
+test('Share Tests', async t => {
+  await t.click(Selector('a'))
+  await t.typeText(page.emailInput, email)
+  await t.typeText(page.passInput, 'Slic123@')
+
+  await t.click(Selector('#signup-btn', { timeout: 1000 }))
+
+  const code = await config.getCode(email)
+  await t.typeText(Selector('#confirmationCode'), code)
+  await t.click(Selector('#confirm-signup-btn'))
+})
+
+test('User can share a list after creation', async t => {
+  await t.typeText(page.emailInput, email)
+  await t.typeText(page.passInput, 'Slic123@')
+
+  await t.expect(page.emailInput.value).contains(email)
+  await t.expect(page.passInput.value).contains('Slic123@')
+  await t.click(page.loginBtn)
+
+  await t.click(Selector('#new-list-button', { timeout: 5000 }))
+
+  const listNameInput = Selector('#name')
+  await t.typeText(listNameInput, 'First List', { timeout: 1000 })
+  await t.expect(listNameInput.value).eql('First List')
+
+  await t.typeText(Selector('#description'), 'List Description')
+  await t.expect(Selector('#description').value).eql('List Description')
+
+  await t.click('#save-btn', { timeout: 2000 })
+  await t.click(Selector('a').withText('First List'))
+  await t.click(Selector('#expansion-summary'))
+  await t.expect(Selector('#list-name').withText('First List')).exists
+  await t.expect(Selector('#list-description').withText('List Description'))
+    .exists
+  const getLocation = ClientFunction(() => document.location.href)
+  await t.expect(getLocation()).contains('/list/')
+
+  await t.click(Selector('#share-list-btn'))
+  const shareEmail = await config.getEmail()
+  await t.typeText(Selector('#email-textfield'), shareEmail)
+  await t.expect(Selector('#email-textfield').value).eql(shareEmail)
+  await t.click('#share-btn')
+  await t.expect(Selector('p').withText('List shared successfully!')).exists
+})
+
+test('User can accept a share request', async t => {
+  const email = config.getEmail()
```

Comment:
We are going through the share process again even though that was covered in the previous test. I think it's okay to use the share link generated in the previous test to avoid the duplication here. Otherwise, extract the share functionality into a separate function so you don't have to copy-paste